packages/alea-frontend/pages/api/homework/update-homework.ts

```diff
@@ -1,30 +1,36 @@
 import { NextApiRequest, NextApiResponse } from 'next';
-import { executeAndEndSet500OnError } from '../comment-utils';
+import { checkIfPostOrSetError, executeAndEndSet500OnError } from '../comment-utils';
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  if (req.method === 'PUT') {
-    const { homeworkId, homeworkName, homeworkDate, courseId, courseInstance, archive, filepath } =
-      req.body;
-
-    if (!homeworkId) {
-      return res.status(400).json({ message: 'homeworkId is required' });
-    }
-
-    try {
-      const result = await executeAndEndSet500OnError(
-        'UPDATE homework SET homeworkName = ?, homeworkDate = ?, courseId = ?, courseInstance = ?, archive = ?, filepath = ? WHERE homeworkId = ?',
-        [homeworkName, homeworkDate, courseId, courseInstance, archive, filepath, homeworkId],
-        res
-      );
-
-      if (result) {
-        res.status(200).json({ message: 'Homework updated successfully!' });
-      }
-    } catch (error) {
-      console.error('Error updating homework:', error);
-      res.status(500).json({ message: 'Error updating homework' });
-    }
-  } else {
-    res.status(405).json({ message: 'Method Not Allowed' });
+  if (!checkIfPostOrSetError(req, res)) return;
+  const {
```

Comment:
Add access control to all homework APIs that only the instructor should access. Create a new ResourceName called 'COURSE_HOMEWORK' (in file resource-action-utils.ts). Also update the variable 'ALL_RESOURCE_TYPES'

See 'packages/alea-frontend/pages/api/set-coverage-timeline.ts' to see how to add check using getUserIdIfAuthorizedOrSetError

Use the new resource (COURSE_HOMEWORK) and MUTATE action.